### PR TITLE
Filter by answer

### PR DESCRIPTION
**Pull request by winged**

### CODE REVIEW

**Comment by winged** [comment]
@czosel what do you think of this, as a first draft?

File: caluma/core/filters.py [hunk 1]
```diff
@@ -217,17 +218,161 @@ def filter(self, qs, value):
...
+        return qs
+
+    def apply_expr(self, qs, expr):
+        question_slug = expr["question"]
+        match_value = expr["value"]
+
+        # hierarchy lookup mode
+        hierarchy = expr.get("hierarchy", AnswerLookupMode.FAMILY)
+        if not hierarchy:
+            hierarchy = AnswerLookupMode.FAMILY
```
Comment (czosel) [inline caluma/core/filters.py]: Isn't this case already covered by the second argument of `get`? (same below)

File: caluma/core/filters.py [hunk 2]
```diff
@@ -217,17 +218,161 @@ def filter(self, qs, value):
...
+        lookup = expr.get("lookup", self.lookup_expr)
+        if not lookup:
+            lookup = self.lookup_expr
+
+        filters = {}
+        if "." in question_slug:
+            split_slug = question_slug.split(".")
+            assert (
+                len(split_slug) == 2
+            ), "Cannot match full slug path, only form_slug.question_slug"
```
Comment (czosel) [inline caluma/core/filters.py]: Nitpick: "full" depends on the form configuration, how about
```suggestion
            ), "Cannot match more than one level of nesting, only form_slug.question_slug"
```

File: caluma/core/filters.py (hunk 1 above)
Comment (winged) [reply caluma/core/filters.py]: get() could return None still, that's why

File: caluma/core/filters.py (hunk 2 above)
Comment (winged) [reply caluma/core/filters.py]: Changed the message a bit

**Pull request merged**